### PR TITLE
creating new signing key when splitting secret from dealer

### PR DESCRIPTION
**Pull request by patnir**
## Summary

Problem: The trusted dealer provides the spending key to create the package. This means that the trusted dealer has the capability to sign the transaction on their own terms without the participants being involved. 

Solution: We generate a separate key that is discarded while partially creating the package (it generates the nullifier key and outgoing view key). This means that the trusted dealer no longer will be able to sign the transaction and only all the participants can do so. 

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

**Review by rohanjadvani — APPROVED** [review]
This looks good to me. I'd wait for an additional approval from @andiflabs

File: ironfish-rust/src/frost_utils/split_secret.rs
Comment (andiflabs) [inline ironfish-rust/src/frost_utils/split_secret.rs:23]: Nit: let's get rid of the name "coordinator"

File: ironfish-rust/src/frost_utils/split_spender_key.rs
Comment (andiflabs) [inline ironfish-rust/src/frost_utils/split_spender_key.rs:96]: Why was this assertion removed?

File: ironfish-rust/src/frost_utils/split_secret.rs
Comment (andiflabs) [inline ironfish-rust/src/frost_utils/split_secret.rs:98]: Why was this removed?

Comment (patnir) [reply ironfish-rust/src/frost_utils/split_secret.rs:98]: because this was removed: 
```
let secret_bytes: [u8; 32] = config
        .secret
        .clone()
        .try_into()
        .map_err(|_| IronfishError::new(IronfishErrorKind::InvalidSecret))?;
```

we can keep the map_err if that is preferred

File: ironfish-rust/src/frost_utils/split_spender_key.rs
Comment (patnir) [reply ironfish-rust/src/frost_utils/split_spender_key.rs]: Added it back, not sure how it got removed. must have made a mistake

File: ironfish-rust/src/frost_utils/split_secret.rs
Comment (patnir) [reply ironfish-rust/src/frost_utils/split_secret.rs:98]: we're no longer passing in a vector. the earlier conversion to the vector was unnecessary. This edge case handled at the napi layer i believe

Comment (patnir) [reply ironfish-rust/src/frost_utils/split_secret.rs:98]: https://github.com/iron-fish/ironfish/blob/rahul/ifl-2307-create-new-gsk-when-splitting-secret-from-dealer/ironfish-rust-nodejs/src/frost.rs#L204

**Review by andiflabs — APPROVED** [review]

**Pull request merged**
